doctr/datasets/detection.py

```diff
@@ -33,11 +33,13 @@ def __init__(
         img_folder: str,
         label_folder: str,
         sample_transforms: Optional[Callable[[Any], Any]] = None,
+        geometric_transforms: Optional[Callable[[Any], Any]] = None,
```

Comment:
for now I suggest only implementation the augmentation without integrating in the datasets. I would argue that the integration should only have up to two types of transforms : sample transforms (before batching), and batch_transforms (after batching).

Currently, the latter being minimal, we haven't implemented that but we might come back to this in the future. Regarding geometric transforms, for the more general purpose of a transformation,  which isn't necessarily geometric, I think we should keep sample_transforms and do the following:
- change expected input/output types of sample_transforms to `img, target` instead of `img`
- for existing transforms applying only to the image, we had a wrapper to take `img, target` as input, but, under-the-hood, only applying a given transform to the img before returning the resulting pair

What do you think?